src/asl.ts

```diff
@@ -868,38 +834,32 @@ export class ASL {
       props.End = true;
     }
     if (isCallExpr(expr)) {
-      const serviceCall = findIntegration(expr);
-      if (serviceCall) {
-        if (
-          isPropAccessExpr(expr.expr) &&
-          (expr.expr.name === "waitFor" || expr.expr.name === "waitUntil")
-        ) {
-          delete (props as any).ResultPath;
-          return <State>{
+      if (isReferenceExpr(expr.expr)) {
+        const ref = expr.expr.ref();
+        if (isIntegration<Integration>(ref)) {
+          const serviceCall = new IntegrationImpl(ref);
+          const taskState = <State>{
             ...serviceCall.asl(expr, this),
             ...props,
           };
-        }
 
-        const taskState = <State>{
-          ...serviceCall.asl(expr, this),
-          ...props,
-        };
-
-        const throwOrPass = this.throw(expr);
-        if (throwOrPass?.Next) {
-          return <State>{
-            ...taskState,
-            Catch: [
-              {
-                ErrorEquals: ["States.ALL"],
-                Next: throwOrPass.Next,
-                ResultPath: throwOrPass.ResultPath,
-              },
-            ],
-          };
+          const throwOrPass = this.throw(expr);
+          if (throwOrPass?.Next) {
+            return <State>{
+              ...taskState,
+              Catch: [
+                {
+                  ErrorEquals: ["States.ALL"],
+                  Next: throwOrPass.Next,
+                  ResultPath: throwOrPass.ResultPath,
+                },
+              ],
+            };
+          } else {
+            return taskState;
+          }
         } else {
-          return taskState;
+          throw Error("");
```

Comment:
What should error message be? Why did the logic change?

Reply:
got rid of `findIntegration` which will align better with validating and handling the async change. Will fix error.